main/src/ca/uwaterloo/flix/language/phase/Resolver.scala

```diff
@@ -2491,18 +2493,21 @@ object Resolver {
   /**
     * Returns the method reflection object for the given `className`, `methodName`, and `signature`.
     */
-  private def lookupJvmMethod(className: String, methodName: String, signature: List[Type], static: Boolean, loc: SourceLocation)(implicit flix: Flix): Validation[Method, ResolutionError] = {
+  private def lookupJvmMethod(className: String, methodName: String, signature: List[Type], retTpe: Type, static: Boolean, loc: SourceLocation)(implicit flix: Flix): Validation[Method, ResolutionError] = {
     // Lookup the class and signature.
     flatMapN(lookupJvmClass(className, loc), lookupSignature(signature, loc)) {
       case (clazz, sig) => try {
         // Lookup the method with the appropriate signature.
         val method = clazz.getDeclaredMethod(methodName, sig: _*)
 
         // Check if the method should be and is static.
-        if (static == Modifier.isStatic(method.getModifiers))
-          method.toSuccess
-        else
+        if (static != Modifier.isStatic(method.getModifiers))
           throw new NoSuchMethodException()
+        // Check that the return type of the method matches the declared type
+        else if (Type.getFlixType(method.getReturnType) != retTpe)
```

Comment:
Lets special case this an only consider Java types to begin with (and maybe primitive types).